cylc/flow/platforms.py

```diff
@@ -298,13 +299,21 @@ def platform_from_job_info(platforms, job, remote):
         ):
             # If we have localhost with a non-background batch system we
             # use the batch system to give a sensible guess at the platform
-            return platform_name
+            output = platform_name
 
         elif (
                 re.fullmatch(platform_name, task_host) and
                 task_job_runner == platform_spec['job runner']
         ):
-            return task_host
+            output = task_host
+
+    if task_host:
+        remote['host'] = task_host
+    if task_job_runner:
+        job['batch system'] = task_job_runner
```

Comment:
Instead of re-adding these to the dict, why not add the platform instead?

Reply:
Apart from the fact that I've agreed to avoid modifying the config object, I'm not sure that I fancy adding a platform permanently to the object config.